### DISCUSSION TITLE
Mobileraker Companion

### DISCUSSION THREAD

**Discussion by Tiwatz** [xblax/flashforge_ad5m_klipper_mod#116] [Q&A]
Anyone who sucessfully installed the Mobileraker companion? https://github.com/Clon1998/mobileraker_companion
The installation requirements does not seem to be an issue.
I tried to run the installer but it runs in Creality K1 mode and the file locations/python version is not the same and it fails.
I am not experienced enough to do a manual installation but maybe someone else has done this and can give some guidance?

**Comment by THEGxHOxST** [comment]
I wouldn't really recommend installing it on the actual printer itself just because there isn't a ton of overhead, But you can install it into an another computer, server or PI pretty easy and just configure it that way, Best part is if you get multiple printers you can just add a to your configuration.

**Comment by consp** [comment]
The install scripts will not work (due to bash requirements, assumption it's running debian etc).

The surefire way to get it working would be to run it somewhere else and connect to moonraker. The pillow requirement may/may not work but if that's the only odd one it might run, but since it needs pillow you'd have to cross compile the virtual env wheels and create a manual virtual env which might be a bridge too far for most.

**Comment by xblax** [comment]
Btw. in the local network Mobileraker does not need the companion app. I already tried this successfully. But I don't see much advantage over just opening Mainsail in the phone browser.

**Comment by Tiwatz** [comment]
> I wouldn't really recommend installing it on the actual printer itself just because there isn't a ton of overhead, But you can install it into an another computer, server or PI pretty easy and just configure it that way, Best part is if you get multiple printers you can just add a to your configuration.

I run the companion on a pi  now and that works fine. Thank you.


> Btw. in the local network Mobileraker does not need the companion app. I already tried this successfully. But I don't see much advantage over just opening Mainsail in the phone browser.

To get notifications I need the companion I believe? Otherwise there is no real benefit over the browser but I like the app :)

**Reply by xblax** [reply]
> To get notifications I need the companion I believe? Otherwise there is no real benefit over the browser but I like the app :)

Yes this might be the case. Probably the companion is needed to send notifications to Google. I did not test it extensively when I tried the app.

**Reply by THEGxHOxST** [reply]
He uses a firebase protocol, it does some pretty complicated stuff on the back end but it seems to work!

**Reply by THEGxHOxST** [reply]
Oh, forgot to mention. If you have trouble with the mobileraker app crashing the printer for whatever reason like me. you might have to remove fluidd from your printer like I did. I could also be very unlucky...
